### PR TITLE
[Portal Generated] Review request for Microsoft.Media to add version 2018-07-01

### PR DESCRIPTION
**Pull request by quintinb**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/mediaservices/resource-manager/Microsoft.Media/stable/2018-07-01/Accounts.json [hunk 1]
```diff
@@ -0,0 +1,910 @@
...
+        "properties": {
+          "$ref": "#/definitions/MetricProperties",
+          "description": "Operation properties format."
+        }
+      },
+      "type": "object",
+      "required": [
+        "name"
+      ]
+    },
```
Comment (anuchandy) [inline specification/mediaservices/resource-manager/Microsoft.Media/stable/2018-07-01/Accounts.json]: Looking at the api, the `Operation` model seems readonly model where its can be only read through `Operations.list`. If so should we mark all properties of this model as `readonly`? 

The new models representing the new properties of Operation are:

`MetricProperties`
`ServiceSpecification`
`Metric`

should we mark properties of all these models are `readonly`? given they are not used else where.

File: specification/mediaservices/resource-manager/Microsoft.Media/stable/2018-07-01/Accounts.json [hunk 2]
```diff
@@ -0,0 +1,910 @@
...
+        "message": {
+          "type": "string",
+          "description": "Specifies the detailed reason if the name is not available."
+        }
+      },
+      "type": "object",
+      "required": [
+        "nameAvailable"
+      ],
+      "description": "The response from the check name availability request."
```
Comment (anuchandy) [inline specification/mediaservices/resource-manager/Microsoft.Media/stable/2018-07-01/Accounts.json]: `EntityNameAvailabilityCheckOutput` represents output of `Locations_CheckNameAvailability` operation, seems all properties can be readonly in this model. right?

File: specification/mediaservices/resource-manager/Microsoft.Media/stable/2018-07-01/AssetsAndAssetFilters.json [hunk 3]
```diff
@@ -0,0 +1,1264 @@
...
+        "streamingLocators": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/AssetStreamingLocator"
+          },
+          "description": "The list of Streaming Locators."
+        }
+      },
+      "type": "object",
+      "description": "The Streaming Locators associated with this asset."
```
Comment (anuchandy) [inline specification/mediaservices/resource-manager/Microsoft.Media/stable/2018-07-01/AssetsAndAssetFilters.json]: are properties of `ListStreamingLocatorsResponse` and `AssetStreamingLocator` eligible to be marked as readonly? they are used to represents result of `Assets.listStreamingLocators` api and not used as payload for any create|update api.

**Comment by anuchandy** [comment]
@quintinb 

1. Is this api-version deployed to production? if no, do you know when we are planning to?
2. please take a look at review comments.

@ravbhatnagar new api-version, please share ARM feedback.

**Comment by ravbhatnagar** [comment]
@KrisBash from ARM to take a look. Kris - we can review this together.

**Comment by KrisBash** [comment]
@quintinb 

Can you schedule a reviewing meeting with the API review board? That would help expedite from the ARM side. Thanks

File: specification/mediaservices/resource-manager/Microsoft.Media/stable/2018-07-01/AccountFilters.json [hunk 4]
```diff
@@ -0,0 +1,553 @@
...
+            ],
+            "modelAsExtensible": true
+          },
+          "description": "The track property condition operation."
+        }
+      },
+      "type": "object",
+      "required": [
+        "property",
+        "operation"
```
[the remark this answers is not in the document]
Comment (quintinb) [reply specification/mediaservices/resource-manager/Microsoft.Media/stable/2018-07-01/AccountFilters.json]: Seems reasonable.  I made the value required as well.

File: specification/mediaservices/resource-manager/Microsoft.Media/stable/2018-07-01/AccountFilters.json [hunk 5]
```diff
@@ -0,0 +1,553 @@
...
+    "FilterTrackSelection": {
+      "properties": {
+        "trackSelections": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/FilterTrackPropertyCondition"
+          },
+          "description": "The track selections."
+        }
+      },
```
[the remark this answers is not in the document]
Comment (quintinb) [reply specification/mediaservices/resource-manager/Microsoft.Media/stable/2018-07-01/AccountFilters.json]: No, you are correct there isn't a scenario where you would provide an empty track selections collection.  I added the required attribute.

File: specification/mediaservices/resource-manager/Microsoft.Media/stable/2018-07-01/AccountFilters.json [hunk 6]
```diff
@@ -0,0 +1,553 @@
...
+          }
+        ],
+        "x-ms-examples": {
+          "Update an Account Filter": {
+            "$ref": "examples/accountFilters-update.json"
+          }
+        }
+      }
+    }
+  },
```
[the remark this answers is not in the document]
Comment (quintinb) [reply specification/mediaservices/resource-manager/Microsoft.Media/stable/2018-07-01/AccountFilters.json]: Correct.  AccountFilters are just DB write operations so there is no need for an LRO.

File: specification/mediaservices/resource-manager/Microsoft.Media/stable/2018-07-01/AssetsAndAssetFilters.json (hunk 3 above)
Comment (quintinb) [reply specification/mediaservices/resource-manager/Microsoft.Media/stable/2018-07-01/AssetsAndAssetFilters.json]: Yes, I marked those as readonly.

File: specification/mediaservices/resource-manager/Microsoft.Media/stable/2018-07-01/Accounts.json [hunk 7]
```diff
@@ -0,0 +1,910 @@
...
+            "values": [
+              {
+                "value": "Average"
+              },
+              {
+                "value": "Count"
+              },
+              {
+                "value": "Total"
+              }
```
[the remark this answers is not in the document]
Comment (quintinb) [reply specification/mediaservices/resource-manager/Microsoft.Media/stable/2018-07-01/Accounts.json]: Added.  The team wanted to show a difference between the count of items (say the number of requests) versus the total of a parameter (total bytes delivered).  These are both summations so it isn't my favorite design but I added some comments to try to make the distinction clearer.

File: specification/mediaservices/resource-manager/Microsoft.Media/stable/2018-07-01/Accounts.json (hunk 1 above)
Comment (quintinb) [reply specification/mediaservices/resource-manager/Microsoft.Media/stable/2018-07-01/Accounts.json]: Yes, done.

**Comment by KrisBash** [comment]
Discussed changes with @quintinb . No further review meeting required from ARM

**Review by KrisBash — APPROVED** [review]
LGTM

File: specification/datalake-analytics/data-plane/readme.nodejs.md [hunk 8]
```diff
@@ -24,4 +25,7 @@ nodejs:
 ``` yaml $(nodejs) && $(tag) == 'package-job'
 nodejs:
   source-code-folder-path: lib/job
-```
\ No newline at end of file
+  generate-license-txt: true
```
Comment (daschult) [inline specification/datalake-analytics/data-plane/readme.nodejs.md]: Revert the changes in this file. I just made these changes and this PR is attempting to undo them.

File: .travis.yml [hunk 9]
```diff
@@ -47,7 +53,7 @@ script:
     fi
   - >-
     if [[ $MODE == 'java' ]]; then
-      travis_wait 30 scripts/swagger-to-sdk.sh Azure/azure-libraries-for-java -v
+      travis_wait 30 scripts/swagger-to-sdk.sh Azure/azure-sdk-for-java -v
```
Comment (daschult) [inline .travis.yml]: I know that azure-libraries-for-java is correct, so this is probably wrong.

**Review by ghost — CHANGES_REQUESTED** [review]

File: specification/datalake-analytics/data-plane/readme.nodejs.md (hunk 8 above)
Comment (quintinb) [reply specification/datalake-analytics/data-plane/readme.nodejs.md]: I think I messed up a merge.  Please stand by...

**Comment by anuchandy** [comment]
Closing this PR in-favor of https://github.com/Azure/azure-rest-api-specs/pull/3980
